src/api/app/models/package.rb

```diff
@@ -308,12 +308,14 @@ def is_locked?
     project.is_locked?
   end
 
+  def master_product_object
+    # test _product permissions if any other _product: subcontainer is used and _product exists
+    product_object = project.packages.find_by(name: "_product") if name =~ /\A_product:\w[-+\w\.]*\z/
```

Comment:
It runs only when write access is checked, but not when you load the object from the database (that was how I understood "package object get's created", right?).

The logic behind is to check the package container "_product" if it exists for write when actually trying to do so for "_product:*". But to check the "_product:*" if it does not exist.

The reason behind is that _product:* are usually create from _product and only that one is authorized to modify it. However, for released products we have only the "_product:*" containers, since they are released independend. So they need to be checked directly.

(Btw, I hope to get that cleaned up with _multibuild & source service for next generation product medias)